client/main.js

```diff
@@ -0,0 +1,470 @@
+import BN from "bn.js";
+import web3Utils from "web3-utils";
+
+const ganache = require("ganache-core");
+const contract = require("truffle-contract");
+const ReputationMiningCycleJSON = require("../build/contracts/ReputationMiningCycle.json");
+const ColonyNetworkJSON = require("../build/contracts/IColonyNetwork.json");
+const PatriciaTreeJSON = require("../build/contracts/PatriciaTree.json");
+
+const ReputationMiningCycle = contract(ReputationMiningCycleJSON);
+
+const jsonfile = require("jsonfile");
+
+const file = "./reputations.json";
+
+const Web3 = require("web3");
+
+const ColonyNetwork = contract(ColonyNetworkJSON);
+const accountAddress = "0xbb46703786c2049d4d6dd43f5b4edf52a20fefe4";
+
+class ReputationMiningClient {
+  constructor(minerAddress, realProviderPort = 8545) {
+    this.web3 = new Web3();
+    this.minerAddress = minerAddress;
+    this.PatriciaTree = contract(PatriciaTreeJSON);
+    this.ganacheProvider = ganache.provider({
+      network_id: 515,
+      vmErrorsOnRPCResponse: false,
+      locked: false,
+      verbose: true,
+      accounts: [
+        {
+          balance: "0x10000000000000000000000000",
+          secretKey: "0xe5c050bb6bfdd9c29397b8fe6ed59ad2f7df83d6fd213b473f84b489205d9fc7"
```

Comment:
Oh, I see it now, this is probably the secret key for the above account address. Still, I think we can make that better.

Reply:
Correct. The mining clients instantiate their own local blockchain and deploy a patricia tree to it so they are using exactly the same code to build their tree as will be used during any dispute resolution that occurs.